lib/jsonpathv2/enumerable.rb

```diff
@@ -64,9 +63,7 @@ def handle_wildecard(node, expr, context, key, pos, &blk)
             next unless start_idx
             end_idx = (array_args[1] && process_function_or_literal(array_args[1], -1) || (sub_path.count(':') == 0 ? start_idx : -1))
             next unless end_idx
-            if start_idx == end_idx
-              next unless start_idx < node.size
-            end
+            next if start_idx == end_idx && start_idx > node.size
```

Comment:
Yeah, I know the feeling. :) Especially when taking over code from somewhere. :D And you sort of rely on the tests to guide you and be there for you while you slowly explore just to find out that they are lying. :D 

![lied](http://i0.kym-cdn.com/photos/images/masonry/000/236/903/7ff.png)